tests/compat/test_gecko47.py

```diff
@@ -0,0 +1,183 @@
+
+        script = '''
+            var certDB = Cc["@mozilla.org/security/x509certdb;1"]
+                                        .getService(Ci.nsIX509CertDB);
+            certdb.importCertsFromFile(null, fp.file, nsIX509Cert.CA_CERT);
+        '''
+
+        self.run_script_for_compat(script)
+
+        assert not self.compat_err.notices
+        assert not self.compat_err.errors
+
+        # There are other checks because using nsIX509CertDB is
+        # potentially dangerous.
+        assert len(self.compat_err.warnings) == 4
+        assert self.compat_err.warnings[3]['compatibility_type'] == 'error'
+        assert self.compat_err.warnings[3]['message'].startswith(expected)
+
+        script = 'const nsIX509CertDB = Components.interfaces.nsIX509CertDB;'
+
+        self.run_script_for_compat(script)
+
+        assert not self.compat_err.notices
+        assert not self.compat_err.errors
+
+        # There are other checks because using nsIX509CertDB is
+        # potentially dangerous.
+        assert len(self.compat_err.warnings) == 3
+        assert self.compat_err.warnings[2]['compatibility_type'] == 'error'
+        assert self.compat_err.warnings[2]['message'].startswith(expected)
+
+    def test_changed_return_type(self):
+        """https://github.com/mozilla/addons-server/issues/2220"""
+        message = (
+            'listTokens() and listSlots() now return '
+            'nsISimpleEnumerator instead of nsIEnumerator.')
+
+        script = 'let tokenList = gTokenDB.listTokens();'
+
+        self.run_script_for_compat(script)
+
+        assert not self.compat_err.notices
+        assert not self.compat_err.errors
+
+        assert len(self.compat_err.warnings) == 1
+        assert self.compat_err.warnings[0]['compatibility_type'] == 'error'
+        assert self.compat_err.warnings[0]['message'].startswith(message)
+
+        script = 'slot = belgiumEidPKCS11Module.listSlots().currentItem();'
+
+        self.run_script_for_compat(script)
+
+        assert not self.compat_err.notices
+        assert not self.compat_err.errors
+
+        assert len(self.compat_err.warnings) == 1
+        assert self.compat_err.warnings[0]['compatibility_type'] == 'error'
+        assert self.compat_err.warnings[0]['message'].startswith(message)
+
+        script = 'let slots = testModule.listSlots();'
+
+        assert not self.compat_err.notices
+        assert not self.compat_err.errors
+
+        assert len(self.compat_err.warnings) == 1
+        assert self.compat_err.warnings[0]['compatibility_type'] == 'error'
+        assert self.compat_err.warnings[0]['message'].startswith(message)
+
+    def test_gDevTools_moved(self):
+        """https://github.com/mozilla/addons-server/issues/2219"""
+        message = 'The gDevTools.jsm module should no longer be used.'
+
+        script = 'Components.utils.import("resource:///modules/devtools/client/framework/gDevTools.jsm", ainspectorSidebar);'
```

Comment:
Looks like there's a bunch over 80. This codebase isn't exactly PEP8 compliant but we should still try to do so with new code IMO. The stuff that's about 100 lines is probably fine but this one is particularly long.